config.go

```diff
@@ -0,0 +1,84 @@
+package x
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+// StoreConfig contains configuration values for the storage engine
+type StoreConfig map[string]string
```

Comment:
Make sure to consider if a method, type, or interface really needs to be exported.

It is not really relevant right now, because there is one package and it is not really intended to be used outside of here, but it is good practice.